dashboard/app/controllers/lti_v1_controller.rb

```diff
@@ -24,17 +25,31 @@ def login
     elsif params[:platform_id]
       query_params = {platform_id: params[:platform_id]}
     else
-      return unauthorized_status
+      return log_unauthorized(
+        "Missing required parameters for LTI authentication",
+        {
+          client_id: params[:client_id],
+          issuer: params[:iss],
+          platform_id: params[:platform_id],
+        }
+      )
     end
 
     lti_integration = LtiIntegration.find_by(query_params)
-    return unauthorized_status unless lti_integration
+    if lti_integration.nil?
+      return log_unauthorized("LTI integration not found", query_params)
+    end
 
     state_and_nonce = create_state_and_nonce
     # set cache key as state value, since we get this back in the final response
     # from the LTI Platform, and can use it to query for these values in the
     # authenticate controller action.
-    write_cache(state_and_nonce[:state], state_and_nonce)
+    begin
+      write_cache(state_and_nonce[:state], state_and_nonce)
+    rescue => exception
+      Honeybadger.notify(exception, context: {message: 'Error writing state and nonce to cache'})
+      return render status: :internal_server_error
```

Comment:
Changed these caching errors to be 500s instead of 401s. If we want to be all-in on plain 401s for all errors in these functions, I can change them back.